### PR TITLE
Add release instructions for Homebrew to README.md

### PR DESCRIPTION
**Pull request by nonrational**
This PR adds instructions for manually authoring a new package suitable for release via a Homebrew formula. Ideally, all of this would be handled by TravisCI, but #babysteps.  In order to support artifact upload via `ghr`, this PR also modifies `make release` to drop binaries into a single directory `./pkg`.

I've make an example of [a release created with these instructions](https://github.com/nonrational/puma-dev/releases/tag/v0.13.nonrational).

Last Homebrew release of `puma-dev` was on [Aug 21, 2018](https://github.com/puma/puma-dev/releases/tag/v0.12). There have been a few updates since then that [seem to fix issues for folks](https://github.com/puma/puma-dev/issues/201#issuecomment-580315723).

### CODE REVIEW

**Comment by nonrational** [comment]
Gotta love it. Appears the TravisCI linux build is failing while trying to `go get ./...`. I tried to replicate the issue on Ubuntu, but didn't get the same error. And macOS build is green.

Looks like this has been broken on master since https://travis-ci.org/puma/puma-dev/builds/609699900?utm_source=github_status&utm_medium=notification but that commit only changed `README.md`. Likely a bug in build config or due to a change at Travis. 😢

File: dev/launch/launch.go [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// +build darwin
```
Comment (nonrational) [inline dev/launch/launch.go]: this file has no business being part of the linux build. it uses `launch_activate_socket` which is only provided on macOS. 

I'm unsure as to why travis picks up this file as part of `go get` but all attempts to replicate locally on linux don't run into the same problem. perhaps there's some greedy compilation configured somewhere? 

either way, this functionality is irrelevant on linux, so it seems safe to isolate this file to builds on darwin arch.

Comment (nonrational) [reply dev/launch/launch.go]: @nateberkopec let me know if you'd prefer that i pr this fix separately.

**Comment by nateberkopec** [comment]
😍 Gorgeous!

File: dev/launch/launch.go (hunk 1 above)
Comment (pda) [reply dev/launch/launch.go]: (for posterity; see also #225)

**Pull request merged**